bellbot_gui/src/gui2.py

```diff
@@ -0,0 +1,282 @@
+            gui_select_dest_header = os.path.join(self.http_root, config_parser.get("gui", "gui_select_dest_header"))
+            gui_select_dest_footer = os.path.join(self.http_root, config_parser.get("gui", "gui_select_dest_footer"))
+            select_dest_page_filename = config_parser.get("gui", "gui_select_dest_page")
+            gui_select_dest_page = os.path.join(self.http_root, select_dest_page_filename)
+        except ConfigParser.NoOptionError:
+            raise
+        with open(gui_select_dest_header, "r") as f:
+            data_gui_select_dest_header = f.read()
+        with open(gui_select_dest_footer, "r") as f:
+            data_gui_select_dest_footer = f.read()
+        options_str = ""
+        for categ, rooms in self.dests_per_categ.items():
+            options_str += '<optgroup label="' + categ + '">\n'
+            rooms_ids = sorted(rooms.keys())
+            for room_id in rooms_ids:
+                options_str += self.ret_options_str(rooms[room_id])
+            options_str += "</optgroup>\n"
+        with open(gui_select_dest_page, "w") as f:
+            f.write(data_gui_select_dest_header)
+            f.write(options_str)
+            f.write(data_gui_select_dest_footer)
+        return select_dest_page_filename
+
+    def ret_options_str(self, room):
+        print(room.name, room.goto)
+        return '<option' + ' id="' + room.id + '" name="' + room.name + '" description="' + room.description + '" kind="' + room.kind + '" goto="' + room.goto + '" available="' + str(room.available) + '" at_node="' + str(room.at_node) +'">' + room.name + "</option>\n"
+        # return '<option' + ' value="' + room.goto + '" id="' + room.id + '" name="' + room.name + '" description="' + room.description + '" kind="' + room.kind + '" goto="' + room.goto + '" available="' + str(room.available) + '" at_node="' + str(room.at_node) +'">' + room.name + "</option>\n"
+
+
+class GUI_Wait_For_Single_Guest(object):
+    def __init__(self):
+        self.service_prefix = '/bellbot_gui_services'
+        self.buttons = [('Go', 'trigger_confirm_single_guest')]
+        rospy.Service(self.service_prefix+'/trigger_confirm_single_guest', std_srvs.srv.Empty, self.trigger_go)
+
+    def display(self, state=None):
+        buttons_content = strands_webserver.page_utils.generate_alert_button_page("", self.buttons, self.service_prefix)
+        message = "<div>Click to confirm that you would like to go to " + state.goal + "</div>" + "<div>" + state.text + "</div>"
+        www_content = message + buttons_content
+        strands_webserver.client_utils.display_content(display_no, www_content)
+
+    def trigger_go(self, req):
+        rospy.wait_for_service('/bellbot_accept_target_single')
+        try:
+            proxy = rospy.ServiceProxy('/bellbot_accept_target_single', NewTarget)
+            res = proxy(NewTargetRequest(""))
+            return res
+        except rospy.ServiceException, e:
+            print "Service call failed: %s"%e
```

Comment:
As mentioned above, service should have a return statement. Please check the other services as well.